subsys/bluetooth/host/gatt.c

```diff
@@ -4473,3 +4355,154 @@ SETTINGS_STATIC_HANDLER_DEFINE(bt_hash, "bt/hash", NULL, db_hash_set,
 			       db_hash_commit, NULL);
 #endif /*CONFIG_BT_GATT_CACHING */
 #endif /* CONFIG_BT_SETTINGS */
+
+static u8_t remove_peer_from_attr(const struct bt_gatt_attr *attr,
+				  void *user_data)
+{
+	const struct addr_with_id *addr_with_id = user_data;
+	struct _bt_gatt_ccc *ccc;
+	struct bt_gatt_ccc_cfg *cfg;
+
+	/* Check if attribute is a CCC */
+	if (attr->write != bt_gatt_attr_write_ccc) {
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	ccc = attr->user_data;
+
+	/* Check if there is a cfg for the peer */
+	cfg = ccc_find_cfg(ccc, addr_with_id->addr, addr_with_id->id);
+	if (cfg) {
+		memset(cfg, 0, sizeof(*cfg));
+	}
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static int bt_gatt_clear_ccc(u8_t id, const bt_addr_le_t *addr)
+{
+	struct addr_with_id addr_with_id = {
+		.addr = addr,
+		.id = id,
+	};
+
+	bt_gatt_foreach_attr(0x0001, 0xffff, remove_peer_from_attr,
+			     &addr_with_id);
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		char key[BT_SETTINGS_KEY_MAX];
+
+		if (id) {
+			char id_str[4];
+
+			u8_to_dec(id_str, sizeof(id_str), id);
+			bt_settings_encode_key(key, sizeof(key), "ccc",
+					       (bt_addr_le_t *)addr, id_str);
```

Comment:
Unrelated to this PR, but it looks like `bt_settings_encode_key()` needs to be fixed to take a const address pointer

Reply:
Actually, I take that back. @Vudentz could you include the following patch as a new commit into this PR, and then fix up your current commit not to do the explicit casts:
```
diff --git a/subsys/bluetooth/host/settings.c b/subsys/bluetooth/host/settings.c
index a97bfe9bec..ea138efaf4 100644
--- a/subsys/bluetooth/host/settings.c
+++ b/subsys/bluetooth/host/settings.c
@@ -21,7 +21,7 @@
 
 #if defined(CONFIG_BT_SETTINGS_USE_PRINTK)
 void bt_settings_encode_key(char *path, size_t path_size, const char *subsys,
-                           bt_addr_le_t *addr, const char *key)
+                           const bt_addr_le_t *addr, const char *key)
 {
        if (key) {
                snprintk(path, path_size,
@@ -41,7 +41,7 @@ void bt_settings_encode_key(char *path, size_t path_size, const char *subsys,
 }
 #else
 void bt_settings_encode_key(char *path, size_t path_size, const char *subsys,
-                           bt_addr_le_t *addr, const char *key)
+                           const bt_addr_le_t *addr, const char *key)
 {
        size_t len = 3;
 
diff --git a/subsys/bluetooth/host/settings.h b/subsys/bluetooth/host/settings.h
index c918ad426c..e4854e437c 100644
--- a/subsys/bluetooth/host/settings.h
+++ b/subsys/bluetooth/host/settings.h
@@ -12,7 +12,7 @@
 
 /* Helpers for keys containing a bdaddr */
 void bt_settings_encode_key(char *path, size_t path_size, const char *subsys,
-                           bt_addr_le_t *addr, const char *key);
+                           const bt_addr_le_t *addr, const char *key);
 int bt_settings_decode_key(const char *key, bt_addr_le_t *addr);
 
 void bt_settings_save_id(void);
```